### ISSUE TITLE
[Proposal] - Similar URL using Named Routes

### ISSUE THREAD

**Issue by prezire** [laravel/ideas#1317]
It would be nice to be able to redirect to similar URLs that are handled by any controllers via different route names.

```php
Route::get('some', 'MyController@some')->name('some');
Route::get('some', 'MyController@thing')->name('thing');

return redirect()->route('some');
return redirect()->route('thing');
```
Perhaps a 2nd parameter can be passed to generate a unique URL for variables.
```php
$some = route('some', true);
$thing = route('thing', true);
```

**Comment by prezire** [comment]
It can use the existing mechanism like use the one at the bottom, which in this case the route named "thing".

**Comment by sisve** [comment]
The name is used when you resolve the route by name in your application. However, when the user visits the `/some/` url there will be no name. How would Laravel know which one of these two routes it should use in that case?

**Comment by thannaske** [comment]
I don't understand your proposal. I neither get the advantage of defining two routes with the exact same URI (not similar as your title says) nor the way you would want to technically implement it.

**Comment by prezire** [comment]
The idea is having the same URIs that are implemented differently. If there are duplicate URIs via URL string (not by route name), then get the last one declared at the bottom of the Route file.

**Comment by jmarcher** [comment]
Just give different routes for different actions, why would someone want to have 5 URIs with the same name and different outcomes? Maybe what you need are Query Parameters.

**Comment by prezire** [comment]
Please re-read multiple examples above.

**Comment by jmarcher** [comment]
It does not make sense to me, if you want to have different actions, then you should have multiple endpoint names or use query parameters in order to determine different actions.
Your examples neither give a proper example nor explains how to make this works nor why this is needed.

**Comment by 36864** [comment]
I have read your examples multiple times. At no point do you make any reference as to how the web server would be able to tell which route to use.

Let's try with an example. Suppose I define the following routes:
```
Route::get('example', 'ExampleController@first')->name('first');
Route::get('example', 'ExampleController@second')->name('second');
Route::get('example', 'ExampleController@third')->name('third');
```

Now the user visits my website at `www.example.com/example`. Which of the three routes does the web server execute?

**Comment by jmarcher** [comment]
@36864 Laravel overwrites the routes by default so the last one will be picked by the server, that is why I see no point on trying to accomplish something like this...

**Comment by fletch3555** [comment]
@36864 I don't believe the proposal is trying to change that.  I believe it's trying to add "name" into the list of fields that differentiate routes, such that a route (`GET /example`) can exist multiple times with different names and different controller/actions.

Using your example, current behavior is that the first 2 are completely ignored and only `third` remains, which is what Laravel's router will find when matching a request for `/example`.  However, generating a url using `route('first')` would fail, but @prezire would like it to succeed.

Having said that, I don't think it'll ever work that way, since the controller actions associated with routes are only ever run when incoming routes are matched, not when matching for URL generation.

@prezire, I think the confusion here is because you haven't provided a use-case.  What are you trying to do that this would help with?  Please be detailed.

**Comment by jmarcher** [comment]
There is no way to implement this, I mean yes, you could get a named route from different names but the Router will still match only one (the last one) of them.
